network-rfc/12_contracts_design.md

```diff
@@ -0,0 +1,90 @@
+# Portal System
+
+
+### Portal Creation and SQD Collection
+
+The Portal System begins when a Data Consumer creates a new portal through the PortalFactory by specifying key parameters including the target amount of SQD tokens needed, deposit deadline, payment token, "budget".
+
+The data consumer allocation (contribution by the deployer) will be determined by the target amount that the data consumer is seeking.
+
+We are collecting 120% of the amount that will be set by SQD.
+
+The factory deploys a single PortalProxy contract, an upgradeable instance that combines both the core distribution logic and SQD vault functionality into one unified contract. 
+Once deployed, SQD token providers can stake their tokens directly into the PortalProxy by calling the stake function with the portal address and desired amount. 
+
+During this collection phase, the portal remains in a "Collecting" state where it accumulates SQD deposits from multiple providers until either the target amount is reached or the deposit deadline passes. 
+If the target is met before the deadline, the data consumer can trigger the activate function to transition the portal to its active distribution phase.
+
+However, if the deadline expires without reaching the target, the portal is marked as failed, triggering a full refund of both the consumer's budget and all staked SQD tokens back to their respective owners.
```

Comment:
What if the deadline is very long? Can providers unlock their funds with the normal exit mechanism? If so, do we even need this cancellation?